src/napari_omero/plugins/loaders.py

```diff
@@ -65,6 +65,34 @@ def omero_proxy_reader(
 def load_image_wrapper(image: ImageWrapper) -> list[LayerData]:
     data = get_data_lazy(image)
     meta = get_omero_metadata(image)
+
+    # check for singleton dims in data to be able to remove them
+    singleton_dims = [dim for dim in range(data.ndim) if data.shape[dim] == 1]
+    # if the channels dim isn't the only singleton, we will squeeze other singletons
+    if not (len(singleton_dims) == 1 and 1 in singleton_dims):
+        if 1 in singleton_dims:  # channels dim
+            # we need to keep this, because we split on it
+            # if it's a singleton, napari will squeeze it out anyways
+            singleton_dims.remove(1)
+        if 0 in singleton_dims:  # time dim
+            # if T is being dropped, update channel_axis for new position of C
+            meta["channel_axis"] = 0
+
+        # squeeze out singleton dims from data
+        data = data.squeeze(axis=tuple(singleton_dims))
+
+        # make sure layer scale and axis_labels are updated for the squeezed dims
+        non_channel_axes = [i for i in range(5) if i != 1]
+        meta["scale"] = [
+            meta["scale"][non_channel_axes.index(i)]
+            for i in range(5)
+            if i not in singleton_dims and i != 1
+        ]
+        meta["axis_labels"] = [
+            meta["axis_labels"][non_channel_axes.index(i)]
+            for i in range(5)
+            if i not in singleton_dims and i != 1
+        ]
```

Comment:
Just for code clarity

```suggestion
        new_scale = []
        new_axis_labels = []
        for i in range(5):
            if i not in singleton_dims and i != 1:
                new_scale.append(meta["scale"][non_channel_axes.index(i)])
                new_axis_labels.append(meta["axis_labels"][non_channel_axes.index(i)])
        meta["scale"] = new_scale
        meta["axis_labels"] = new_axis_labels
```

Reply:
I think readability here is in the eye of the beholder?
The logic is just reversed so you see what is being set prior to the conditions.
list comprehensions are pretty Pythonic and my understanding more performant (although here it's just range(5) so no difference).

(Which reminds me, we might want to revisit some of the other nested loops and look into map_blocks, etc.)

I can be swayed either way.

Reply:
I agree with the performance and Pythonic-part. I was more thinking about less proficient code readers :) 

Feel free to change or not 👍 Otherwise, I'll just go ahead and merge.

Reply:
Fair enough -- @jo-mueller I wonder, are you a polyglot? code in other languages than just Python?
Because as someone that only really touches Python and isn't super proficinet -- is **far** from an expert -- the 
```
meta["axis_labels"] = [
            meta["axis_labels"][non_channel_axes.index(i)]
            ...
```
structure is more natural and intuitive. I like seeing what is set to what like this.
But I expect nested for/if to be universal in almost all languages?

Anyhow, I prefer to leave it! 😜

Reply:
Sounds alright to me 

As for your question: I'm 99% Python. If I can, I use list comprehensions wherever I go but I also do a fair bit of teaching and I try to not push it all the way in this context ^^"